test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -112,14 +112,14 @@ def do_h(q: int):
     h(q)
 
 
-@aq.function
+@aq.function(num_qubits=6)
 def recursive_h(q: int):
     do_h(q)
     if q > 0:
         recursive_h(q - 1)
 
 
-@aq.function
+@aq.function(num_qubits=6)
```

Comment:
Ugh. This is almost a counterexample to this change, because the way this program is written, the `q` argument has to be correlated with `num_qubits`, and now we have split that to two locations in the code. This is a contrived test case so I wouldn't overindex on it, just something I hadn't thought of before.

Reply:
Yeah, I also noticed that :/ I still think this is an improvement, but we'll have to be on the lookout for real world examples that run into friction with this style

Reply:
Is it a typical use case to have wrapper functions like this? It seems like either could be the "main" if you split up the concepts of main function and subroutine. Customers would be back to needing to make sure that they sync between all of them.

Reply:
What do you mean by sync between all of them? Splitting up the main/subroutine concepts would mean that these tests would have to be updated, because they would not be correct as written. Specifically:

```
@aq.subroutine  # num_qubits not accepted, cannot generate recursive_h directly
def recursive_h(q: int):
    do_h(q)
    if q > 0:
        recursive_h(q - 1)

@aq.main(num_qubits=6)
def recursive_h_wrapper():
    q = aq.get_num_qubits()  # showing for demonstration, Ryan made this suggestion
    recursive_h(q)

program = recursive_h_wrapper()
```

Reply:
I add `(num_qubits=6)` to both of them here because the tests call each of them, and that minimizes the changes to the tests. An end user should not have to call both of them -- these examples are pretty contrived as Ryan said.